### DISCUSSION TITLE
example of using GHC 9.4 or other version

### DISCUSSION THREAD

**Discussion by peterbecich** [srid/haskell-template#87] [Q&A] [answered]
Thank you for this project and `haskell-flake`!

Would you provide an example of overriding the default GHC version in `nixpkgs-unstable` from 9.2 to 9.4, etc.?

I am not sure how to use these:
- [haskellPackages.ghc_9_4_4](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=ghc_9_4)
- [haskell.compiler.ghc943](https://search.nixos.org/packages?channel=unstable&show=haskell.compiler.ghc943&from=0&size=50&sort=relevance&type=packages&query=ghc94)

**Comment by srid** [comment, marked as answer]
You want to set `basePackages` to the appropriate package set:

https://flake.parts/options/haskell-flake.html#opt-perSystem.haskellProjects._name_.basePackages

For eg.,

```nix
basePackages = pkgs.haskell.packages.ghc943;
```

**Reply by peterbecich** [reply]
great, it works, thanks

**Reply by wolverian** [reply]
Hmm, this doesn't actually work for me anymore. For some reason upstream Nixpkgs asserts graphql like this:

```
error: assertion '((lib).versionOlder (self).hspec.version "2.10")' failed
       at /nix/store/88fpp9lrbs0vk8yf94bz045y3bfz7m2r-source/pkgs/development/haskell-modules/configuration-common.nix:290:5:
          289|   graphql =
          290|     assert lib.versionOlder self.hspec.version "2.10";
             |     ^
          291|     doJailbreak super.graphql;
```

**Reply by wolverian** [reply]
(I don't use graphql so this is annoying.)

**Reply by wolverian** [reply]
I found a possible related PR, but I don't understand it: https://github.com/NixOS/nixpkgs/pull/166425

**Reply by billksun** [reply]
I just hit the same issue (with the graphql assertion).
